spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/MessageChannelBinderSupport.java

```diff
@@ -82,13 +80,17 @@
 public abstract class MessageChannelBinderSupport
 		implements Binder<MessageChannel>, ApplicationContextAware, InitializingBean {
 
-	protected static final String P2P_NAMED_CHANNEL_TYPE_PREFIX = "queue:";
-
-	protected static final String PUBSUB_NAMED_CHANNEL_TYPE_PREFIX = "topic:";
+	protected static final String PARTITION_HEADER = "partition";
 
-	protected static final String JOB_CHANNEL_TYPE_PREFIX = "job:";
+	/**
+	 * Default group name (used if <code>null</code> or empty String is provided).
+	 */
+	protected static final String DEFAULT_CONSUMER_GROUP = "default";
```

Comment:
This is not really effective in the context of SCS, where the default is to set a UUID when no group property is provided for a binding. That has been done to fix https://github.com/spring-cloud/spring-cloud-stream/issues/211, and I think we should preserve it - applications wishing to act as competing consumers should rather explicitly register as part of a group than the other way around (apps acting as competing consumers unless they are registered as part of distinct consumer groups). I think we should disable the current defaulting mechanism, and put in place an alternative as part of upcoming refactoring.